### ISSUE TITLE
Method to keep the god rays effect visible even when the light source is behind camera

### ISSUE THREAD

**Issue by juliangarnier** [pmndrs/postprocessing#361]
Hi,

I'm trying to amplify the lighting effect of a spotlight with a god rays effect.
It works great when the spotlight is facing the camera, but fails when either the spotlight is outside the camera Frustum or behind an other object.

I don't expect this effect to work when the spotlight is behind an other object in the scene, but I was wondering if it would be possible to prevent the effect from disappearing completely when the light source is outside of the camera frustum.

Basically to prevent this scenario when the effect still working when the light source is on the edge of the screen
<img width="1032" alt="Screenshot 2022-04-01 at 13 02 12" src="https://user-images.githubusercontent.com/1268691/161251355-e14442a1-f97d-4702-8143-81fda0ed15a4.png">

But disappears as soon as it leaves the camera Frustum
<img width="1031" alt="Screenshot 2022-04-01 at 13 02 24" src="https://user-images.githubusercontent.com/1268691/161251377-4ff0804b-f15c-434a-a60b-568c5d4257ed.png">

I noticed that [the example explicitly set](https://github.com/pmndrs/postprocessing/blob/daba8ea41d4d45857658e39d60fce9fe3c940848/demo/src/demos/GodRaysDemo.js#L193) `sun.frustumCulled = false` which I thought was for this purpose, but doesn't appear to have any effect, at least regarding this problem.

Any input on how to keep the god rays visible when the light source is outside the camera frustum?

Thank you.

**Comment by vanruesc** [comment]
Hi,

> Any input on how to keep the god rays visible when the light source is outside the camera frustum?

Unfortunately, the `GodRaysEffect` can only blur light sources that are actually visible on screen. The `frustumCulled = false` bit is only really needed when using a `Points` light source to ensures that it doesn't suddenly disappear when near the screen edges.

True volumetric lighting requires a different approach, but I haven't really looked into alternatives yet. Unreal, for example, seems to use particles for its [volumetric fog](https://docs.unrealengine.com/4.27/en-US/BuildingWorlds/FogEffects/VolumetricFog/), so a postprocessing filter may never achieve the same results. If someone has any resources on the matter or knows of other approaches, please let me know.

**Comment by juliangarnier** [comment]
Hi,

> Unfortunately, the GodRaysEffect can only blur light sources that are actually visible on screen. The frustumCulled = false bit is only really needed when using a Points light source to ensures that it doesn't suddenly disappear when near the screen edges.

Thanks for the explanation, I couldn't figure out what the `frustumCulled = false` was actually doing and gave me false hope that it could have an effect of my issue.

I managed to get the spot light effect I was looking for by using a cone instead of a sphere geometry for the light source.
So instead of trying to emulate the light bulb with a small sphere, I use the code to recreate the base of the light beam.

Closing the issue, will try to post some results here once the website is live.
